contracts/modules/andromeda-curve/src/contract.rs

```diff
@@ -116,50 +139,26 @@ pub fn execute_update_curve_config(
         .add_attribute("sender", sender))
 }
 
-pub fn execute_update_restriction(
-    ctx: ExecuteContext,
-    restriction: CurveRestriction,
-) -> Result<Response, ContractError> {
-    nonpayable(&ctx.info)?;
-    let sender = ctx.info.sender;
-    ensure!(
-        ADOContract::default().is_owner_or_operator(ctx.deps.storage, sender.as_ref())?,
-        ContractError::Unauthorized {}
-    );
-    RESTRICTION.save(ctx.deps.storage, &restriction)?;
-
-    Ok(Response::new()
-        .add_attribute("method", "update_restriction")
-        .add_attribute("sender", sender))
-}
-
-pub fn execute_reset(ctx: ExecuteContext) -> Result<Response, ContractError> {
+pub fn execute_reset(mut ctx: ExecuteContext) -> Result<Response, ContractError> {
     nonpayable(&ctx.info)?;
     let sender = ctx.info.sender.clone();
-    ensure!(
-        has_permission(ctx.deps.storage, &sender)?,
-        ContractError::Unauthorized {}
-    );
+    ADOContract::default().is_permissioned(
+        ctx.deps.branch(),
+        ctx.env.clone(),
+        RESET_ACTION,
+        sender.clone(),
+    )?;
 
     CURVE_CONFIG.remove(ctx.deps.storage);
 
     Ok(Response::new().add_attribute("method", "reset"))
 }
 
-pub fn has_permission(storage: &dyn Storage, addr: &Addr) -> Result<bool, ContractError> {
-    let is_operator = ADOContract::default().is_owner_or_operator(storage, addr.as_str())?;
-    let allowed = match RESTRICTION.load(storage)? {
-        CurveRestriction::Private => is_operator,
-        CurveRestriction::Public => true,
-    };
-    Ok(is_operator || allowed)
-}
-
 #[cfg_attr(not(feature = "library"), entry_point)]
 pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> Result<Binary, ContractError> {
     match msg {
         QueryMsg::GetCurveConfig {} => encode_binary(&query_curve_config(deps.storage)?),
-        QueryMsg::GetRestriction {} => encode_binary(&query_restriction(deps.storage)?),
+        // QueryMsg::GetRestriction {} => encode_binary(&query_restriction(deps.storage)?),
```

Comment:
Should be deleted instead of commented out

Reply:
Oh, resolved